core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java

```diff
@@ -153,6 +164,21 @@ public void loadCarbonTable(TableInfo tableInfo) {
         tableInfo.getFactTable().getBucketingInfo());
   }
 
+  private void parseSortColumns(TableSchema tableSchema) {
+    Map<String, String> tableProperties = tableSchema.getTableProperties();
+    if (tableProperties != null) {
+      String sortColumnsString = tableProperties.get(CarbonCommonConstants.SORT_COLUMNS);
+      if (sortColumnsString != null) {
+        numberOfSortColumns = sortColumnsString.split(",").length;
+        for (int i = 0; i < numberOfSortColumns; i++) {
+          if (!tableSchema.getListOfColumns().get(i).hasEncoding(Encoding.DICTIONARY)) {
+            numberOfNoDictSortColumns++;
```

Comment:
during table creation,  moved sort columns to the begin of table schema by the order of sort columns.